framework_tvb/tvb/adapters/datatypes/db/connectivity.py

```diff
@@ -61,11 +64,21 @@ class ConnectivityIndex(DataType):
     def fill_from_has_traits(self, datatype):
         # type: (Connectivity)  -> None
         super(ConnectivityIndex, self).fill_from_has_traits(datatype)
+
+        self.has_cortical_mask = True
+        if datatype.cortical is None or len(set(datatype.cortical)) != 2:
```

Comment:
why `len(set(datatype.cortical)) != 2`? If all elements of `.cortical` are `True`, it is still a cortical mask.  I am surprised that `.cortical is None` is insufficient.  It would be a lot cleaner to write
```python
self.has_cortical_mask = datatype.cortical is not None
```

Reply:
You are right @maedoc this was wrongly covering another problem documented by Paula today: when `Conn.cortical` was all True (so only cortical regions) in the SpatialMonitor it was converted in a `spatial_mask` of only value 1, which was incompatible with the current monitor code (which requires `spatial_mask` to have values starting from 0.
It should be fixed now.